### PR TITLE
Update server

### PR DESCRIPTION
**Pull request by zellman01**
- [X] Add EXP to chat to allow users to get EXP
- [X] Show the creator of the specific events in the view command for room-events
- [X] Eslint the above changes
- [X] Update Dev Dependencies
- [X] Allow users in a private faction to see that faction's profile
- [X] Allow Upper-Staff to see private faction profiles
- [X] Test changes
- [X] Eslint remaining things
- [X] Add Insanity back to formats
- [X] Add remaining legendaries to Insanity mod
- [X] Alert staff members when PMing someone that has ignored pms that the user has ignored pms
- [ ] Get shadowban to work (failed tests. Will work on later)
- [X] Test modifications
- [X] Eslint changes

If there are any more things that I will try to change, I will change this PR's description

### CODE REVIEW

File: chat.js [hunk 1]
```diff
@@ -288,13 +288,41 @@ class CommandContext {
...
+					for (let u in this.room.users) {
+						let curUser = Users(u);
+						if (!curUser || !curUser.connected) continue;
+						if (OCPU.ignoreEmotes[curUser.userid]) {
+							curUser.sendTo(this.room, (this.room.type === 'chat' ? '|c:|' + (~~(Date.now() / 1000)) + '|' : '|c|') + this.user.getIdentity(this.room.id) + '|' + message);
+							continue;
+						}
+						curUser.sendTo(this.room, (this.room.type === 'chat' ? '|c:|' + (~~(Date.now() / 1000)) + '|' : '|c|') + this.user.getIdentity(this.room.id) + '|/html ' + emoticons);
+					}
+					this.room.log.log.push((this.room.type === 'chat' ? (this.room.type === 'chat' ? '|c:|' + (~~(Date.now() / 1000)) + '|' : '|c|') : '|c|') + this.user.getIdentity(this.room.id) + '|' + message);
```
Comment (DeathlyPlays) [inline chat.js]: Bug: This actually doesn't log the message.
I know that from my own experience with this lol

Comment (DeathlyPlays) [reply chat.js]: Not major but will need to be fixed

Comment (zellman01) [reply chat.js]: Well, I had about 3 minutes to actually get that, so yeah. I didn't have time to look over it. What would fix it? (Also, I need to Eslint it)

Comment (DeathlyPlays) [reply chat.js]: idk how to fix it yet havent had time

**Comment by zellman01** [comment]
That should be the last Eslint in Chat (until I change it again)

**Comment by zellman01** [comment]
Okay. Just review it

File: chat-commands.js [hunk 2]
```diff
@@ -438,6 +438,9 @@ exports.commands = {
 			connection.send(error);
 			return;
 		}
+		if (targetUser.ignorePMs === true && this.can('lock')) {
```
Comment (AlphaWind) [inline chat-commands.js]: Doesn't this exist though? For staff ignoring PMs

Comment (zellman01) [reply chat-commands.js]: That is just to alert the staff member that the specific user is ignoring PMs. It still gets sent.

Comment (AlphaWind) [reply chat-commands.js]: Alright everything else looks good to me

**Pull request merged**